### PR TITLE
scripts for building libmxnet binary and wheel

### PR DESCRIPTION
**Pull request by szha**
## Description ##
scripts for building libmxnet binary and wheel

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] add script for making all dependencies
- [x] tools for building pip package
- [x] build scripts for lib and wheel

### CODE REVIEW

**Comment by roywei** [comment]
@mxnet-label-bot add[Build, pr-awaiting-review]

File: tools/build/build_lib.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+fi
+
+git clone --recursive https://github.com/apache/incubator-mxnet mxnet-build
+
+>&2 echo "Now building mxnet modules..."
+cp $make_config mxnet-build/config.mk
+
+cd mxnet-build
+
+$MAKE DEPS_PATH=$DEPS_PATH DMLCCORE
```
Comment (lanking520) [inline tools/build/build_lib.sh]: Please specify the `$MAKE`

**Comment by byronyi** [comment]
This is insensible. 

Why do you hard-code git clone to mxnet-build when you already have a source tree to start with? 

What if I try to maintain a fork and doesn't want to do nested checkout to the upstream?

**Comment by szha** [comment]
@byronyi I imagine you can replace the clone command with a symlink to the parent folder

**Comment by szha** [comment]
@byronyi we also didn't expect creating a wheel to be a prerequisite of submitting a PR. If there's any specific issue we can help with, feel free to create an issue.

**Comment by byronyi** [comment]
Symlink does the trick. I will take a closer look and see how to fix this.

And you’re right, building a wheel is not necessary before a PR. But it’s also makes little value for us to send out a PR without first trying it out, which usually requires a wheel package in our in-house environment. I thought it’s a common requirement for certain users, but I could be wrong.

It would be best if we could simply produce nightly packages in the same way you produce one and release it on PyPI. Currently I’m not sure how to do that.

**Comment by szha** [comment]
@byronyi I agree that simplifying it would add value, and a fix is definitely more than welcome. This PR is a simple migration of code from a separate repo and hence the current structure.

**Comment by byronyi** [comment]
Sorry, symlink does not appear to work.

**Pull request merged**